core/src/dird/ua_prune.cc

```diff
@@ -741,7 +704,7 @@ static bool PruneBackupJobs(UaContext* ua,
    * in the configuration file. Interesting ClientId/FileSetId will be
    * added to jobids_check.
    */
-  if (!ua->db->SqlQuery(query.c_str(), JobSelectHandler, jobids_check)) {
+  if (!ua->db->SqlQuery(query.c_str(), JobSelectHandler, jobids_check.get())) {
```

Comment:
I have no idea. 
Did not even notice it, did not check JobSelectHandler as I only needed to change the pointer to a smart one. 
I'll take a look and see, my guess is that it has some historic reasons and most probably one of them was changed but the other was forgotten, as the compiler cannot detect it because of the void pointer thing.